In [ ]:
import pandas as pd

#The dataset includes training and testing splits typically used for predictive modeling. However, since this task focuses on exploratory analysis, only the training dataset was used.

df = pd.read_csv("customer_churn_dataset-training-master.csv")
df.head()

,CustomerID,Age,Gender,Tenure,Usage Frequency,Support Calls,Payment Delay,Subscription Type,Contract Length,Total Spend,Last Interaction,Churn
0,2.0,30.0,Female,39.0,14.0,5.0,18.0,Standard,Annual,932.0,17.0,1.0
1,3.0,65.0,Female,49.0,1.0,10.0,8.0,Basic,Monthly,557.0,6.0,1.0
2,4.0,55.0,Female,14.0,4.0,6.0,18.0,Basic,Quarterly,185.0,3.0,1.0
3,5.0,58.0,Male,38.0,21.0,7.0,7.0,Standard,Monthly,396.0,29.0,1.0
4,6.0,23.0,Male,32.0,20.0,5.0,8.0,Basic,Monthly,617.0,20.0,1.0


In [2]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 440833 entries, 0 to 440832
Data columns (total 12 columns):
 #   Column             Non-Null Count   Dtype  
---  ------             --------------   -----  
 0   CustomerID         440832 non-null  float64
 1   Age                440832 non-null  float64
 2   Gender             440832 non-null  str    
 3   Tenure             440832 non-null  float64
 4   Usage Frequency    440832 non-null  float64
 5   Support Calls      440832 non-null  float64
 6   Payment Delay      440832 non-null  float64
 7   Subscription Type  440832 non-null  str    
 8   Contract Length    440832 non-null  str    
 9   Total Spend        440832 non-null  float64
 10  Last Interaction   440832 non-null  float64
 11  Churn              440832 non-null  float64
dtypes: float64(9), str(3)
memory usage: 40.4 MB


In [3]:
df.isnull().sum()

CustomerID           1
Age                  1
Gender               1
Tenure               1
Usage Frequency      1
Support Calls        1
Payment Delay        1
Subscription Type    1
Contract Length      1
Total Spend          1
Last Interaction     1
Churn                1
dtype: int64

In [4]:
df.columns

Index(['CustomerID', 'Age', 'Gender', 'Tenure', 'Usage Frequency',
       'Support Calls', 'Payment Delay', 'Subscription Type',
       'Contract Length', 'Total Spend', 'Last Interaction', 'Churn'],
      dtype='str')

In [5]:
df.shape

(440833, 12)

# Data Exploration

The dataset contains ~440K customer records with behavioral, financial, and subscription features, along with a churn indicator. Missing values are negligible (only one per column) and do not impact analysis.

In [8]:
df.dropna(inplace=True)

In [9]:
df['Churn'].value_counts()

Churn
1.0    249999
0.0    190833
Name: count, dtype: int64

In [10]:
churn_counts = df['Churn'].value_counts(normalize=True) * 100
churn_counts

Churn
1.0    56.71072
0.0    43.28928
Name: proportion, dtype: float64

# Churn Proportion

Approximately 56.7% of customers have churned, while 43.3% remain active. This indicates a relatively high churn rate, suggesting potential issues with retention or customer experience.

In [12]:
num_cols = df.select_dtypes(include=['float64', 'int64']).columns
correlation = df[num_cols].corr()['Churn'].sort_values(ascending=False)
correlation

Churn               1.000000
Support Calls       0.574267
Payment Delay       0.312129
Age                 0.218394
Last Interaction    0.149616
Usage Frequency    -0.046101
Tenure             -0.051919
Total Spend        -0.429355
CustomerID         -0.839365
Name: Churn, dtype: float64

In [13]:
df.groupby('Contract Length')['Churn'].mean()

Contract Length
Annual       0.460761
Monthly      1.000000
Quarterly    0.460256
Name: Churn, dtype: float64

In [14]:
df.groupby('Subscription Type')['Churn'].mean()

Subscription Type
Basic       0.581782
Premium     0.559417
Standard    0.560700
Name: Churn, dtype: float64

In [15]:
df.groupby('Gender')['Churn'].mean()

Gender
Female    0.666691
Male      0.491269
Name: Churn, dtype: float64

# Key Features Correlated with Churn

**Support Calls** show the strongest positive correlation with churn, indicating that customers requiring frequent support are more likely to leave.

**Contract Length** has a significant impact, with customers on monthly contracts showing extremely high churn compared to longer-term contracts.

**Total Spend** is negatively correlated with churn, suggesting that higher-value customers are less likely to leave.

# Churn Intervention Strategy

Customers with high support interaction and those on monthly contracts show the highest likelihood of churn. A targeted intervention would be to proactively identify such customers and offer improved support experiences along with incentives to shift to longer-term contracts.

For example, customers with frequent support calls can be provided priority support or issue resolution follow-ups, while monthly users can be encouraged to switch to annual plans through discounts or added benefits.

In [16]:
results = df[['CustomerID', 'Churn', 'Support Calls', 'Payment Delay', 'Total Spend', 'Contract Length']]
results.to_csv("churn_analysis_results.csv", index=False)